server/server.go

```diff
@@ -22,31 +22,38 @@ var logger = monitor.NewModuleLogger("server")
 type Server struct {
 	monitor.ModuleLogger
 
-	InterruptChan  chan os.Signal
 	DefaultHeaders map[string]string
 	Environment    *environment.Env
 	ProxyService   *proxy.ProxyService
 
+	stopChan chan os.Signal
```

Comment:
We have a stopper pattern in lbry.go https://github.com/lbryio/lbry.go/tree/master/extras/stop

You may find it useful. I use it in both internal-apis and chainquery. It's pretty useful.

Reply:
I looked at it but felt like it's an overkill here. Plus we don't have direct control over whatever goroutines are spawned during the server life cycle.